prow/repoowners/repoowners.go

```diff
@@ -215,36 +217,58 @@ func (c *Client) LoadRepoOwners(org, repo, base string) (RepoOwner, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	entry, ok := c.cache[fullName]
-	if !ok || entry.sha != sha || entry.owners == nil || entry.owners.enableMDYAML != mdYaml {
+	if !ok || entry.sha != sha || !entry.isValid(mdYaml) {
 		gitRepo, err := c.git.Clone(cloneRef)
 		if err != nil {
 			return nil, fmt.Errorf("failed to clone %s: %v", cloneRef, err)
 		}
 		defer gitRepo.Clean()
-		if err := gitRepo.Checkout(base); err != nil {
-			return nil, err
-		}
 
-		if entry.aliases == nil || entry.sha != sha {
-			// aliases must be loaded
-			entry.aliases = loadAliasesFrom(gitRepo.Dir, log)
+		reusable := entry.isValid(mdYaml)
+		// In most sha changed cases, the files associated with the owners are unchanged.
+		// The cached entry can continue to be used, so need do git diff
+		if reusable {
```

Comment:
Please add some unit tests for this functionality. This looks right to me, but this is a critical plugin used by a lot of repos so I'd hate to see it break.

Reply:
fixed